service/src/main/java/de/starwit/service/impl/ServiceInterface.java

```diff
@@ -18,23 +18,28 @@
  *
  * @param <E>
  */
-public interface ServiceInterface<E extends AbstractEntity<Long>> {
+public interface ServiceInterface<E extends AbstractEntity<Long>, R extends JpaRepository<E, Long>> {
 
 	static Logger LOG = LoggerFactory.getLogger(ServiceInterface.class);
 
-	public void delete(Long id) throws EntityNotFoundException;
-
-	public E saveOrUpdate(E entity) throws ValidationException;
-	
-	public List<E> findAll();
-
-	public E findById(Long id);
-	
-	public default E create(E entity) throws ValidationException {
-		return saveOrUpdate(entity);
-	}
-	
-	public default E update(E entity) throws ValidationException {
-		return saveOrUpdate(entity);
-	}
+	public R getRepository();
+
+	default public List<E> findAll() {
+        return this.getRepository().findAll();
+    }
+
+    default public E findById(Long id) {
+        return this.getRepository().findById(id).orElseThrow(() -> new EntityNotFoundException(String.valueOf(id)));
```

Comment:
Maybe an identifier could be helpful of what could not be found. E.g. ServiceInterface_ID